src/adapter/src/coord.rs

```diff
@@ -2081,13 +2098,79 @@ pub async fn serve(
     let (bootstrap_tx, bootstrap_rx) = oneshot::channel();
     let handle = TokioHandle::current();
 
-    let initial_timestamps = catalog.get_all_persisted_timestamps().await?;
     let metrics = Metrics::register_into(&metrics_registry);
     let metrics_clone = metrics.clone();
+    let timestamp_oracle_metrics =
+        Arc::new(timestamp_oracle::metrics::Metrics::new(&metrics_registry));
     let segment_client_clone = segment_client.clone();
     let span = tracing::Span::current();
     let coord_now = now.clone();
     let advance_timelines_interval = tokio::time::interval(catalog.config().timestamp_interval);
+
+    // We get the timestamp oracle impl once on startup, to ensure that it
```

Comment:
Let's pull this whole block out into a method :)

Reply:
good call! I've gotten too used to methods being several pages long :see_no_evil: